tools/fs/optimistic.ts

```diff
@@ -273,6 +274,24 @@ export const optimisticLStatOrNull = makeCheapPathFunction(
   },
 );
 
+export const optimisticRealpath = makeOptimistic('realpath', realpath);
+export const optimisticRealpathOrNull = makeOptimistic('realpathOrNull', (
+  path: string,
+  options?: Parameters<typeof optimisticRealpath>[1],
+) => {
+  try {
+    return realpath(path, options)
+  } catch (e) {
+    if (e.code !== "ENOENT") {
+      throw e;
+    }
+  }
+
+  dependOnParentDirectory(path);
```

Comment:
Yeah, I'm afraid `realpath` depends on more information than just the `path` and the parent directory. We could reimplement it here, using `optimisticLStatOrNull` instead of `fs.lstat`, but that would be trickier.

Reply:
I've removed the optimisticRealpath since it was only used for two things: the realpath of each node_modules folder, which doesn't have much of an impact on build performance, and for symlinks in `node_modules/.bin`, which there normally are none Windows.